sharktuner/tests/constraint_generator_test.py

```diff
@@ -181,17 +181,21 @@ def test_generate_solutions_tile_and_fuse_contraction_padding(
 
         assert len(solutions) > 0, "No solutions generated with TileAndFuse pipeline."
         assert all(
-            isinstance(sol, iree_codegen.CompilationInfoAttr) for sol in solutions
-        )
+            len(pair) == 1 and pair[0][0] == "compilation_info" for pair in solutions
+        ), "Each solution must be a single-item list with key 'compilation_info'"
 
+        compilation_infos = [pair[0][1] for pair in solutions]
+        assert all(
+            isinstance(sol, iree_codegen.CompilationInfoAttr)
+            for sol in compilation_infos
+        )
         assert all(
-            "padding =" in str(sol.lowering_config) for sol in solutions
+            "padding =" in str(sol.lowering_config) for sol in compilation_infos
         ), "Not all lowering configs have padding option."
-
         assert all(
             [int(x) for x in sol.lowering_config.attributes["promote_operands"]]
             == [0, 1, 2]
-            for sol in solutions
+            for sol in compilation_infos
         ), "Not all lowering configs have promote_operands = [0, 1, 2]."
```

Comment:
I think we could simplify this code ba bunch by writing a for loop and then checking each pair separately (instead of relying on `all`). Once you name the pair elements the code will become more readable.